coalib/core/Bear.py

```diff
@@ -0,0 +1,501 @@
+
+    To specify additional attributes to your bear, use the following:
+
+    >>> class SomeBear(Bear):
+    ...     AUTHORS = {'Jon Snow'}
+    ...     AUTHORS_EMAILS = {'jon_snow@gmail.com'}
+    ...     MAINTAINERS = {'Catelyn Stark'}
+    ...     MAINTAINERS_EMAILS = {'catelyn_stark@gmail.com'}
+    ...     LICENSE = 'AGPL-3.0'
+    ...     ASCIINEMA_URL = 'https://asciinema.org/a/80761'
+
+    If the maintainers are the same as the authors, they can be omitted:
+
+    >>> class SomeBear(Bear):
+    ...     AUTHORS = {'Jon Snow'}
+    ...     AUTHORS_EMAILS = {'jon_snow@gmail.com'}
+    >>> SomeBear.maintainers
+    {'Jon Snow'}
+    >>> SomeBear.maintainers_emails
+    {'jon_snow@gmail.com'}
+
+    If your bear needs to include local files, then specify it giving strings
+    containing relative file paths to the INCLUDE_LOCAL_FILES set:
+
+    >>> class SomeBear(Bear):
+    ...     INCLUDE_LOCAL_FILES = {'checkstyle.jar', 'google_checks.xml'}
+
+    To keep track easier of what a bear can do, simply tell it to the
+    ``CAN_FIX`` and the ``CAN_DETECT`` sets. Possible values are:
+
+    >>> CAN_DETECT = {'Syntax', 'Formatting', 'Security', 'Complexity',
+    ... 'Smell', 'Unused Code', 'Redundancy', 'Variable Misuse', 'Spelling',
+    ... 'Memory Leak', 'Documentation', 'Duplication', 'Commented Code',
+    ... 'Grammar', 'Missing Import', 'Unreachable Code', 'Undefined Element',
+    ... 'Code Simplification'}
+    >>> CAN_FIX = {'Syntax', ...}
+
+    Specifying something to ``CAN_FIX`` makes it obvious that it can be
+    detected too, so it may be omitted:
+
+    >>> class SomeBear(Bear):
+    ...     CAN_DETECT = {'Syntax', 'Security'}
+    ...     CAN_FIX = {'Redundancy'}
+    >>> sorted(SomeBear.can_detect)
+    ['Redundancy', 'Security', 'Syntax']
+
+    Every bear has a data directory which is unique to that particular bear:
+
+    >>> class SomeBear(Bear): pass
+    >>> class SomeOtherBear(Bear): pass
+    >>> SomeBear.data_dir == SomeOtherBear.data_dir
+    False
+
+    A bear can be dependant from other bears. ``BEAR_DEPS`` contains bear
+    classes that are to be executed before this bear gets executed. The results
+    of these bears will then be passed inside ``self.dependency_results`` as a
+    dict. The dict will have the name of the bear as key and a list of its
+    results as values:
+
+    >>> class SomeBear(Bear): pass
+    >>> class SomeOtherBear(Bear):
+    ...     BEAR_DEPS = {SomeBear}
+    >>> SomeOtherBear.BEAR_DEPS
+    {<class 'coalib.core.Bear.SomeBear'>}
+    """
+
+    LANGUAGES = set()
+    REQUIREMENTS = set()
+    AUTHORS = set()
+    AUTHORS_EMAILS = set()
+    MAINTAINERS = set()
+    MAINTAINERS_EMAILS = set()
+    PLATFORMS = {'any'}
+    LICENSE = ''
+    INCLUDE_LOCAL_FILES = set()
+    CAN_DETECT = set()
+    CAN_FIX = set()
+    ASCIINEMA_URL = ''
+    BEAR_DEPS = set()
+
+    @classproperty
+    def name(cls):
+        """
+        :return:
+            The name of the bear.
```

Comment:
Any reason for keeping it?
As soon as the docstyle bear is finished (I hope someday...), we switch to this format anyway^^